src/core/varlink-unit.c

```diff
@@ -0,0 +1,420 @@
+        // Mount/Automount context
+        // Path context
+        // Scope context
+        // Swap context
+        // Timer context
+        // Service context
+        // Socket context
+}
+
+static int can_clean_build_json(sd_json_variant **ret, const char *name, void *userdata) {
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        Unit *u = ASSERT_PTR(userdata);
+        ExecCleanMask mask;
+        int r;
+
+        assert(ret);
+
+        r = unit_can_clean(u, &mask);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to check if unit can be cleaned: %m");
+
+        for (ExecDirectoryType t = 0; t < _EXEC_DIRECTORY_TYPE_MAX; t++) {
+                if (!BIT_SET(mask, t))
+                        continue;
+
+                r = sd_json_variant_append_arrayb(&v, SD_JSON_BUILD_STRING(exec_resource_type_to_string(t)));
+                if (r < 0)
+                        return r;
+        }
+
+        if (FLAGS_SET(mask, EXEC_CLEAN_FDSTORE)) {
+                r = sd_json_variant_append_arrayb(&v, SD_JSON_BUILD_STRING("fdstore"));
+                if (r < 0)
+                        return r;
+        }
+
+        *ret = TAKE_PTR(v);
+        return 0;
+}
+
+static int markers_build_json(sd_json_variant **ret, const char *name, void *userdata) {
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        unsigned *markers = ASSERT_PTR(userdata);
+        int r;
+
+        assert(ret);
+
+        BIT_FOREACH(m, *markers) {
+                r = sd_json_variant_append_arrayb(&v, SD_JSON_BUILD_STRING(unit_marker_to_string(m)));
+                if (r < 0)
+                        return r;
+        }
+
+        *ret = TAKE_PTR(v);
+        return 0;
+}
+
+static int activation_details_build_json(sd_json_variant **ret, const char *name, void *userdata) {
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        const ActivationDetails *activation_details = userdata;
+        _cleanup_strv_free_ char **pairs = NULL;
+        int r;
+
+        assert(ret);
+
+        // activation_details_append_pair() gracefully takes activation_details==NULL
+        r = activation_details_append_pair(activation_details, &pairs);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get activation details: %m");
+
+        STRV_FOREACH_PAIR(key, value, pairs) {
+                r = sd_json_variant_append_arraybo(&v,
+                                SD_JSON_BUILD_PAIR_STRING("type", *key),
+                                SD_JSON_BUILD_PAIR_STRING("name", *value));
+                if (r < 0)
+                        return r;
+        }
+
+        *ret = TAKE_PTR(v);
+        return 0;
+}
+
+static int unit_runtime_build_json(sd_json_variant **ret, const char *name, void *userdata) {
+        Unit *u = ASSERT_PTR(userdata);
+        Unit *f = unit_following(u);
+
+        return sd_json_buildo(ASSERT_PTR(ret),
```

Comment:
please break line directly after sd_json_buildo(

(also apply to varlink-manager.c)